src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java

```diff
@@ -141,4 +141,14 @@ private void updateTransactionWithLinks(Transaction transaction,
     private String getSubmissionUri(String transactionId, String submissionId) {
         return String.format(SUBMISSION_URI_PATTERN, transactionId, submissionId);
     }
+
+    public ResponseEntity<Object> deleteAcspApplication(String userId){
+        Optional<AcspDataDao> acspData = acspRepository.findById(userId);
+        if (acspData.isPresent()) {
+            acspRepository.deleteById(userId);
```

Comment:
If delete is successful then also delete the corresponding transaction. You can most probably delete from here only